### PR TITLE
remove initial_farming_amount from farm token attr

### PR DESCRIPTION
**Pull request by psorinionut**

### CODE REVIEW

**Review by CostinCarabas — APPROVED** [review]

File: common/common_structs/src/farm_types.rs [hunk 1]
```diff
@@ -81,8 +76,6 @@ pub trait FarmToken<M: ManagedTypeApi> {
     fn get_reward_per_share(&self) -> BigUint<M>;
 
     fn get_compounded_rewards(&self) -> BigUint<M>;
-
-    fn get_initial_farming_tokens(&self) -> BigUint<M>;
```
Comment (dorin-iancu) [inline common/common_structs/src/farm_types.rs:85]: Don't remove this method from the trait.

**Review by dorin-iancu — CHANGES_REQUESTED** [review]

File: farm-staking/farm-staking-proxy/src/lp_farm_token.rs [hunk 2]
```diff
@@ -22,7 +22,7 @@ pub trait LpFarmTokenModule: token_merge_helper::TokenMergeHelperModule {
         self.rule_of_three_non_zero_result(
             farm_token_amount,
             &attributes.current_farm_amount,
-            &attributes.initial_farming_amount,
+            &(&attributes.current_farm_amount - &attributes.compounded_reward),
```
Comment (dorin-iancu) [inline farm-staking/farm-staking-proxy/src/lp_farm_token.rs:25]: Just use the `get_initial_farming_tokens` from the `FarmToken` trait.

**Review by dorin-iancu — CHANGES_REQUESTED** [review]

File: common/common_structs/src/farm_types.rs (hunk 1 above)
Comment (psorinionut) [reply common/common_structs/src/farm_types.rs]: Done.

File: farm-staking/farm-staking-proxy/src/lp_farm_token.rs (hunk 2 above)
Comment (psorinionut) [reply farm-staking/farm-staking-proxy/src/lp_farm_token.rs]: Done.

**Review by dorin-iancu — APPROVED** [review]

**Pull request merged**